### ISSUE TITLE
IllegalStateException when using viewBinding for onCreateView

### ISSUE THREAD

**Issue by gabrielfeo** [androidbroadcast/ViewBindingPropertyDelegate#101]
Trying to update to 1.5.2+, I find this usage causes a crash due to [a new internal check in the library](https://github.com/androidbroadcast/ViewBindingPropertyDelegate/blob/1.5.6/vbpd/vbpd-noreflection/src/main/java/by/kirich1409/viewbindingdelegate/FragmentViewBindings.kt#L83).

```kotlin
    private val binding by viewBinding {
        OnboardingContainerFragmentBinding.inflate(layoutInflater)
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?,
    ): View = binding.root
```

```
IllegalStateException: Fragment’s view can’t be accessed. Fragment’s view is null.
Maybe you try to access view before onViewCreated() or after onDestroyView().
Add check `if (view != null)` before call ViewBinding
```

[The change is stated in the changelog](https://github.com/androidbroadcast/ViewBindingPropertyDelegate/blob/1.5.6/CHANGELOG.md#viewbindingproperty-will-throw-error-when-it-will-be-created-before-host-ready), but I don't see anything inherently wrong with the above usage considering the viewBinding lambda is invoked lazily. Do you see a way to keep it (we have many in the project) when upgrading to 1.5.2+?

**Comment by kirich1409** [comment]
Please try version 1.5.6

**Comment by ddello32** [comment]
I've have been facing the same issue with version 1.5.6

**Comment by kirich1409** [comment]
@gabrielfeo from what places you access viewBinding delegate?

**Comment by cmorigaki** [comment]
@kirich1409 I could replicate the problem using the sample app adjusting the way we setup the Fragment view. Instead of referecing the xml, using the Binding generated class

from:
```
class ProfileFragment : Fragment(R.layout.fragment_profile) {

    private val viewBinding by viewBinding(FragmentProfileBinding::bind,
        onViewDestroyed = { _: FragmentProfileBinding ->
            // reset view
        })
```
to:
```
class ProfileFragment : Fragment() {

    private val viewBinding by viewBinding(FragmentProfileBinding::bind,
        onViewDestroyed = { _: FragmentProfileBinding ->
            // reset view
        })

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ) = viewBinding.root
```

**Comment by rpavliuk** [comment]
Is it possible to fix this without removing the XML referencing?

**Comment by rpavliuk** [comment]
> @gabrielfeo from what places you access viewBinding delegate?

In my case, I'm accessing the `viewBinding` delegate in the `onCreateView` method. Is this ok?

**Comment by kirich1409** [comment]
> 

Yes. `onCreateView()` only need to inflate view from XML or create in code. Only after the callback method Fragment.getView() will return View connected with Fragment. That why all view initialization must be done in `onViewCreated` independently are you using ViewBinding or not.

**Comment by kirich1409** [comment]
Another variant is
```kotlin
class ProfileFragment : Fragment() {
    private val viewBinding: ProfileBinding by viewBinding(createMethod = CreateMethod.INFLATE)

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ) = viewBinding.root
}
```

it will work because ViewBindingPropdetyDelegate will call ViewBinding.inflate method via reflection. It's slower and not recommended.

Prefer to use
```kotlin
class ProfileFragment : Fragment(R.layout.fragment) {
    private val viewBinding: ProfileBinding by viewBinding(ProfileBinding::bind
}
```

**Comment by kirich1409** [comment]
@gabrielfeo can you check the latest version ?

**Comment by kirich1409** [comment]
By default the library prevent using `viewBinding` delegate before onViewCreated() will return the result. That's why crash happens. It's expected behaviour. I will think about @gabrielfeo case will work without turning off the check

**Comment by mahmud0v** [comment]
@kirich1409 Crash happened on android 13, 14 version which is using viewBinding delegate . I used this type

`class ProfileFragment : Fragment(R.layout.profile) {
    private val viewBinding: ProfileBinding by viewBinding(ProfileBinding::bind)
}`

I replaced this code part it worked: 

`class ProfileFragment : Fragment(R.layout.profile) {
    private var _binding: ProfileBinding? = null
    private val binding get() = _binding!!

   override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
    ): View? {
        _binding = ProfileBinding.inflate(inflater, container, false)
        return binding.root
    }
}
`

**Comment by fearsom03** [comment]
any updates on this issue ?

**Comment by mahmud0v** [comment]
> any updates on this issue ?
I don't have any info about this update. Do you have the same issue when using viewBinding delegate with 13, 14 android version

**Comment by kirich1409** [comment]
I understand the issue. Try to reproduce it and fix on weekend

**Comment by kirich1409** [comment]
As solution I recommend to set XML layout in Fragment constructor and use ViewBindingDelegate without without inflating view and it will wrap Fragment.getView()

**Comment by kirich1409** [comment]
Will be fixed in 2.0.0
